hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTableResultScanner.java

```diff
@@ -143,6 +144,18 @@ private void resumePrefetch() {
     resumer = null;
   }
 
+  private void termianteResumerIfPossible() {
+    if (resumer == null) {
+      return;
+    }
+    if (resumer instanceof ScanResumerImpl) {
+      ((ScanResumerImpl) resumer).terminate();
```

Comment:
Better add some comments here to explain why we have a cast here, instead of introduing a method directly in ScanResumer.